src/api_routes_v4.rs

```diff
@@ -0,0 +1,392 @@
+          .route("/report/list", get().to(list_pm_reports)),
+      )
+      // User
+      .service(
+        scope("/account/auth")
+          .guard(guard::Post())
+          .wrap(rate_limit.register())
+          .route("/register", post().to(register))
+          .route("/login", post().to(login))
+          .route("/logout", post().to(logout))
+          .route("/password_reset", post().to(reset_password))
+          .route("/get_captcha", get().to(get_captcha))
+          .route("/password_change", post().to(change_password_after_reset))
+          .route("/change_password", put().to(change_password))
+          .route("/totp/generate", post().to(generate_totp_secret))
+          .route("/totp/update", post().to(update_totp))
+          .route("/verify_email", post().to(verify_email)),
+      )
+      .route("/account/settings/save", put().to(save_user_settings))
+      .service(
+        scope("/account/settings")
+          .wrap(rate_limit.import_user_settings())
+          .route("/export", get().to(export_settings))
+          .route("/import", post().to(import_settings)),
+      )
+      .service(
+        scope("/account")
+          .route("/my_user", get().to(get_my_user))
+          .route("/list_media", get().to(list_media))
+          .route("/mention", get().to(list_mentions))
+          .route("/replies", get().to(list_replies))
+          .route("/delete", post().to(delete_account))
+          .route(
+            "/mention/mark_as_read",
+            post().to(mark_person_mention_as_read),
+          )
+          .route(
+            "/mention/mark_all_as_read",
+            post().to(mark_all_notifications_read),
+          )
+          .route("/report_count", get().to(report_count))
+          .route("/unread_count", get().to(unread_count))
+          .route("/list_logins", get().to(list_logins))
+          .route("/validate_auth", get().to(validate_auth))
+          .service(
+            scope("/block")
+              .route("/person", post().to(user_block_person))
+              .route("/community", post().to(user_block_community))
+              .route("/instance", post().to(user_block_instance)),
+          ),
+      )
+      // User actions
+      .route("/person", get().to(read_person))
+      // Admin Actions
+      .service(
+        scope("/admin")
+          .route("/add", post().to(add_admin))
+          .route(
+            "/registration_application/count",
+            get().to(get_unread_registration_application_count),
+          )
+          .route(
+            "/registration_application/list",
+            get().to(list_registration_applications),
+          )
+          .route(
+            "/registration_application/approve",
+            put().to(approve_registration_application),
+          )
+          .route(
+            "/registration_application",
+            get().to(get_registration_application),
+          )
+          .route("/list_all_media", get().to(list_all_media))
```

Comment:
Both "/account/list_media" and "/admin/list_all_media" should be moved to something like "/media/list"

Reply:
Then how to distinguish between admin and user endpoints? They need to be separate because admins may also want to view their personal uploads.

Reply:
Either one works for me, but I think I like the `/admin/list_all_media` a little better than `/media/list_all`

Reply:
The `/media/list` endpoint can have an optional person id filter.